pkg/controllers/vdb/restart_reconcile.go

```diff
@@ -390,16 +391,33 @@ func (r *RestartReconciler) execRestartPods(ctx context.Context, downPods []*Pod
 	metrics.NodesRestartDuration.With(labels).Observe(elapsedTimeInSeconds)
 	metrics.NodesRestartAttempt.With(labels).Inc()
 	if err != nil {
-		r.VRec.Event(r.Vdb, corev1.EventTypeWarning, events.NodeRestartFailed,
-			"Failed while calling 'admintools -t restart_node'")
 		metrics.NodesRestartFailed.With(labels).Inc()
+		r.logATFailureEvent("restart_node", stdout)
 		return stdout, err
 	}
 	r.VRec.Eventf(r.Vdb, corev1.EventTypeNormal, events.NodeRestartSucceeded,
 		"Successfully called 'admintools -t restart_node' and it took %ds", int(elapsedTimeInSeconds))
 	return stdout, nil
 }
 
+// logATFailureEvent will log k8s events for the admintools restart error.
+func (r *RestartReconciler) logATFailureEvent(atCmd, op string) {
```

Comment:
Are you going to use this for other admintools commands? Like if there is a need to have the same kind of function for another reconciler.

Reply:
There is a bunch of parsing we do for create_db. I debated about throwing everything into a new package that they both use. Do you think I should do that cleanup? It will probably make things easier when we move away from admintools.

Reply:
Yes I do like that idea.

Reply:
Are you going to do that on this PR?

Reply:
Yep, I'll include it in this PR. I'm working on this change now.